src/LanguageServer/Protocol/RoslynRequestExecutionQueue.cs

```diff
@@ -25,15 +27,24 @@ public RoslynRequestExecutionQueue(AbstractLanguageServer<RequestContext> langua
             _initializeManager = languageServer.GetLspServices().GetRequiredService<IInitializeManager>();
         }
 
-        public override Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool rethrowExceptions)
+        public override async Task WrapStartRequestTaskAsync(Task nonMutatingRequestTask, bool rethrowExceptions)
         {
             if (rethrowExceptions)
             {
-                return nonMutatingRequestTask;
+                try
+                {
+                    await nonMutatingRequestTask.ConfigureAwait(false);
+                }
+                // If we had an exception, we want to record a NFW for it AND propogate it out to the queue so it can be handled appropriately.
+                catch (Exception ex) when (FatalError.ReportAndPropagateUnlessCanceled(ex, ErrorSeverity.Critical))
```

Comment:
i believe this is the correct pattern to report and throw, but let me know if not.